2020-02-Spring/3_trees/DeepestLeavesSum/DeepestLeavesSum.cpp

```diff
@@ -0,0 +1,78 @@
+#include <iostream>
+#include <queue>
+using namespace std;
+
+class BTree;
+struct TreeNode;
+
+struct TreeNode {
+  int val;
+  TreeNode *left;
+  TreeNode *right;
+  TreeNode() : val(0), left(nullptr), right(nullptr) {}
+  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
+  TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
+};
+
+static int deepestLeavesSum(TreeNode* root) {
+
+  queue<TreeNode*> queue;
+  queue.push(root);                         // we insert the root into the queue
+  int currSum = 0;                          //will store the sum of the nodes at the current level
+
+  while(!queue.empty()){
+    currSum = 0;                           // reset the sum at each new level of the tree
+    int size = queue.size();              // locking the size of the queue before we add more nodes to it
+
+    //for the size of the queue
+    for(int i =0; i< size;i++){
+      auto node = queue.front();
+      // Pop one node of the queue
+      queue.pop();
+
+      currSum += node->val;
+      //get left child into queue
+      if(node->left !=nullptr){
+
+        queue.push(node->left);
+      }
+      //get right child into queue
+      if(node->right !=nullptr){
+        queue.push(node->right);
+      }
+
+    }
+  }
+  return currSum;
+}
+
+int main() {
+  /**
```

Comment:
It might help to add a comment showing what this tree looks like so it's easier to see what's being tested. :)

Something like this:

```
  3
 / \
1   4
   /
  2
```